src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -157,18 +167,21 @@ private boolean matches(Offer offer, NMProfile profile) {
         if (mem < 0) {
             LOGGER.error("No mem resource present");
         }
+        if (ports < 0) {
+            LOGGER.error("No port resources present");
+        }
 
         Map<String, String> requestAttributes = new HashMap<>();
 
         if (taskUtils.getAggregateCpus(profile) <= cpus
                 && taskUtils.getAggregateMemory(profile) <= mem
                 && SchedulerUtils.isMatchSlaveAttributes(offer,
-                requestAttributes)) {
+                requestAttributes) && ports >= 3) {
```

Comment:
It might be better to move the magic number into the relevant class to which it belongs. For e.g:
```
public class NMPorts {
  private static final String NM_RPC_PORT_KEY = "nm.rpc.port";
  private static final String NM_LOCALIZER_PORT_KEY = "nm.localizer.port";
  private static final String NM_WEBAPP_PORT_KEY = "nm.webapp.port";
  private static final String NM_HTTP_SHUFFLE_PORT_KEY = "nm.http.shuffle.port";

  private static final String NM_PORT_KEYS = 
                 new String[] {
                     NM_RPC_PORT_KEY,
                     NM_LOCALIZER_PORT_KEY,
                     NM_WEBAPP_PORT_KEY,
                     NM_HTTP_SHUFFLE_PORT_KEY
                  };

  private Map<String, Long> nmPorts = 
                  new HashMap<String, Long>(NM_PORT_KEYS.size());

  public NMPorts(long[] ports) { // takes an array rather than args
      if (ports.length != NM_PORT_KEYS.length) {
          // throw error 
      }
      for (int i = 0; i < ports.length; i++) {
         nmPorts.put(NM_PORT_KEYS[i], ports[i]);
      }
  }
 
  // the magic number is replaced by a static getter
  public static int getNumPorts() {
     return nmPorts.size();
  }
  
  public long getRpcPort() {
     return nmPorts.get(NM_RPC_PORT_KEY);
  }
  // getters for other ports 
}
```